apis/policies/v1alpha2/clusteradmissionpolicy_types.go

```diff
@@ -157,21 +157,39 @@ const (
 	// PolicyServerConfigMapReconciled represents the condition of the
 	// Policy Server ConfigMap reconciliation
 	PolicyServerConfigMapReconciled PolicyConditionType = "PolicyServerConfigMapReconciled"
-	// PolicyServerWebhookConfigurationReconciled represents the
-	// condition of the Policy Server WebhookConfiguration
-	// reconciliation
-	PolicyServerWebhookConfigurationReconciled PolicyConditionType = "PolicyServerWebhookConfigurationReconciled"
+	// ClusterAdmissionPolicyActive represents the condition of the Policy
+	// admission webhook being registered
+	ClusterAdmissionPolicyActive PolicyConditionType = "active"
+)
+
+// +kubebuilder:validation:Enum=unscheduled;unschedulable;pending;active
+type ClusterAdmissionPolicyStatusEnum string
+
+const (
+	// ClusterAdmissionPolicyStatusUnscheduled is a transient state that will continue
+	// to unschedulable or pending. This is the default state.
+	ClusterAdmissionPolicyStatusUnscheduled ClusterAdmissionPolicyStatusEnum = "unscheduled"
+	// ClusterAdmissionPolicyStatusUnschedulable informs that policy server where to
+	// schedule the policy is not available
+	ClusterAdmissionPolicyStatusUnschedulable ClusterAdmissionPolicyStatusEnum = "unschedulable"
+	// ClusterAdmissionPolicyStatusPending informs that the policy server exists,
+	// we are reconciling all resources
+	ClusterAdmissionPolicyStatusPending ClusterAdmissionPolicyStatusEnum = "pending"
+	// ClusterAdmissionPolicyStatusActive informs that the k8s API server should be
+	// forwarding admission review objects to the policy
+	ClusterAdmissionPolicyStatusActive ClusterAdmissionPolicyStatusEnum = "active"
 )
 
 // ClusterAdmissionPolicyStatus defines the observed state of ClusterAdmissionPolicy
 type ClusterAdmissionPolicyStatus struct {
-	// PolicyActive represents whether this AdmissionPolicy is active,
-	// such that the Kubernetes API server should be forwarding
-	// admission review objects to the policy
-	PolicyActive bool `json:"policyActive"`
+	// PolicyStatus represents whether this ClusterAdmissionPolicy is unscheduled,
+	// unschedulable, pending, or active.
+	// +kubebuilder:default:=unscheduled
```

Comment:
we can remove default from here since it's not working

Reply:
I would prefer not to, it ends up in the CRD, and then users are aware that policies start in that state (even if it is set by the controller).

Reply:
I also think is misleading. A `kubebuilder:default` marker has a very specific behavior that one can take for granted. Given this is not having the expected behavior (the api machinery is not really persisting a defaulted status field) I think is better to not include it here. We are using our own means for "defaulting" it on the reconciliation phase, but that has no relationship with this marker.

Reply:
Fair enough then :).
Reverted commit in https://github.com/kubewarden/kubewarden-controller/pull/76/commits/84a55a2754bd4b53cca50d942b7245547b52d6b6.